shepherd/shepherd/shepherd.py

```diff
@@ -188,65 +206,105 @@ def _slaughter_sheep(self, sheep_id: str) -> None:
                 logging.warning('Failed to check sheep\'s health '  # pragma: no cover
                                 'due to the following exception: %s', str(se))
 
+    async def _prepare_job_dir(self, sheep, job_id) -> None:
+        """
+        Prepare the working directory for a job (clean it and download inputs).
+
+        :param sheep: The sheep assigned to the job
+        :param job_id: The job for which we are preparing the directory
+        """
+        logging.info('Preparing working directory for job `%s` on `%s`', job_id, sheep.id)
+        working_directory = create_clean_dir(path.join(sheep.sheep_data_root, job_id))
+        await self._storage.pull_job_data(job_id, working_directory)
+        create_clean_dir(path.join(working_directory, OUTPUT_DIR))
+
+    async def _reconfigure_sheep(self, sheep: BaseSheep, job_id: str, model: ModelModel) -> bool:
+        """
+        Slaughter a sheep and resurrect it with a different model.
+
+        :param sheep: the sheep to be reconfigured
+        :param job_id: ID of the job that prompted the reconfiguration
+        :param model: the new model
+        :return: whether or not the reconfiguration succeeded
+        """
+
+        # we need to wait for the in-progress jobs which might already be in the socket
+        async with self.job_done_condition:
+            await self.job_done_condition.wait_for(lambda: len(sheep.in_progress) == 0)
+
+        self._slaughter_sheep(sheep.id)
+
+        try:
+            self._start_sheep(sheep.id, model.name, model.version)
+            return True
+        except SheepConfigurationError as sce:
```

Comment:
The `except` clauses looks quite similar, can we join them or is there a good reason to keep them separated?